### PR TITLE
MINOR: more flexibility for fields to be included

### PR DESCRIPTION
**Pull request by sunnysideup**
If accepted I will clean up messages, etc...

### CODE REVIEW

**Comment by michalkleiner** [comment]
I think I have a rough idea of what you're trying to achieve but can you elaborate what is the actual purpose of the addition or add a docblock or similar?
I also think you mean allowed relation types but inline you call it restrictions which seems more like exclusions, so the opposite.

**Comment by sunnysideup** [comment]
Have another look now - is that better?

File: src/ProvidesHasOneInlineFields.php [hunk 1]
```diff
@@ -9,16 +9,39 @@
...
      */
-    public function provideHasOneInlineFields($relationName)
+    public function provideHasOneInlineFields(string $relationName, ?array $fieldsToShow = [])
     {
-        $fields = $this->getCMSFields()->dataFields();
+        // work out which fields to show
+        $finalFieldsToShow = [];
+        if (count($fieldsToShow) > 0) {
+            foreach ($fieldsToShow as $fieldsToShowEntry) {
+                $relList = array_keys($this->Config()->get($fieldsToShowEntry));
```
Comment (michalkleiner) [inline src/ProvidesHasOneInlineFields.php:28]:
```suggestion
                $relList = array_keys($this->config()->get($fieldsToShowEntry));
```

**Comment by michalkleiner** [comment]
Looks better now, thanks. I can't merge it myself, has to be @stevie-mayhew, but I'd give it 👍.

**Comment by stevie-mayhew** [comment]
I trust you Michal 😀

**Pull request merged**